src/main/java/net/glowstone/inventory/GlowMetaSkull.java

```diff
@@ -93,24 +120,44 @@ public boolean setOwner(String name) {
         if (owner == null) {
             return false;
         }
-        this.owner = owner;
+        this.owner.set(owner);
         return true;
     }
 
     @Override
     public OfflinePlayer getOwningPlayer() {
-        return ((GlowServer) Bukkit.getServer()).getOfflinePlayer(owner);
+        return ((GlowServer) Bukkit.getServer()).getOfflinePlayer(owner.get());
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * When this returns false, it may still succeed asynchronously.
+     */
     @Override
     public boolean setOwningPlayer(OfflinePlayer owningPlayer) {
         if (hasOwner()) {
             return false;
         }
+        return setOwningPlayerInternal(owningPlayer);
+    }
+
+    private boolean setOwningPlayerInternal(OfflinePlayer owningPlayer) {
         if (owningPlayer instanceof GlowOfflinePlayer) {
             GlowOfflinePlayer impl = (GlowOfflinePlayer) owningPlayer;
-            this.owner = impl.getProfile();
+            this.owner.set(impl.getProfile());
+            return true;
+        } else {
```

Comment:
`owningPlayer` could also be a `GlowPlayer` (which does not extend `GlowOfflinePlayer`).

Reply:
Handled that case.